libyul/backends/evm/StackLayoutGenerator.cpp

```diff
@@ -493,6 +527,15 @@ list<pair<CFG::BasicBlock const*, CFG::BasicBlock const*>> StackLayoutGenerator:
 			},
 			[&](CFG::BasicBlock::FunctionReturn const&) {},
 			[&](CFG::BasicBlock::Terminated const&) {},
+			[&](CFG::BasicBlock::Switch const& _switch)
+			{
+				// TODO: Confirm that there are no backwards jumps here
+				if (_switch.defaultCase != nullptr)
+					_addChild(_switch.defaultCase);
+				for (auto const& [caseValue, caseBlock]: _switch.cases)
+					_addChild(caseBlock);
+				_addChild(_switch.target);
```

Comment:
Just to note it here again: it's only possible to directly jump to the target here, if there is no default case, so if there is a default case, ``_switch`` should not contain a ``target``.
Alternatively, you could reuse the same field in ``Switch`` for both and call it ``defaultCaseOrAfterSwitch`` (or just ``defaultCase`` with a comment that it will point to the block after the switch, in case there was no explicit default case).

Reply:
The reason I stored the two separately to begin with is that in the most optimized jump table, the default case block must come before all the other case blocks, since offsets from the default case block are stored in the jump table. But I now realize that it's also possible to arrange the jump table the other way around, i.e. to require the default case block to come after all the other case blocks and instead of storing a positive offset, we store negative offsets in the jump table. This way, we can neatly keep the default case block and `target` together in the way you suggested.

Reply:
I actually hadn't considered that, but yeah :-)!